src/rotation.zig

```diff
@@ -320,6 +320,193 @@ pub const RotationModes = struct {
 			return true;
 		}
 	};
+	pub const Branch = struct { // MARK: Branch
+		pub const id: []const u8 = "branch";
+		pub const dependsOnNeighbors = true;
+		var branchModels: std.StringHashMap(u16) = undefined;
+		const BranchData = packed struct(u6) {
+			isConnectedNegX: bool,
+			isConnectedPosX: bool,
+			isConnectedNegY: bool,
+			isConnectedPosY: bool,
+			isConnectedNegZ: bool,
+			isConnectedPosZ: bool,
+		};
+
+		fn init() void {
+			branchModels = .init(main.globalAllocator.allocator);
+		}
+
+		fn deinit() void {
+			branchModels.deinit();
+		}
+
+		fn branchTransform(quad: *main.models.QuadInfo, data: BranchData) void {
+			for(&quad.corners) |*corner| {
+				if(!data.isConnectedNegX and corner[0] == 0) {
+					degenerateQuad(quad);
+					break;
+				}
+				if(!data.isConnectedPosX and corner[0] == 1) {
+					degenerateQuad(quad);
+					break;
+				}
+				if(!data.isConnectedNegY and corner[1] == 0) {
+					degenerateQuad(quad);
+					break;
+				}
+				if(!data.isConnectedPosY and corner[1] == 1) {
+					degenerateQuad(quad);
+					break;
+				}
+				if (!data.isConnectedNegZ and corner[2] == 0) {
+					degenerateQuad(quad);
+					break;
+				}
+				if (!data.isConnectedPosZ and corner[2] == 1) {
+					degenerateQuad(quad);
+					break;
```

Comment:
You could also say `return degenerateQuat(quad)` to save some lines here.
(This works because both return values are void)